File: java/spring-arch-unit/spring-arch-unit-mvc-example/src/main/java/com/bmarques/springarchunitmvcexample/domain/invoice/ElectronicInvoice.java
package com.bmarques.springarchunitmvcexample.domain.invoice;

import com.bmarques.springarchunitmvcexample.domain.customer.Customer;

import java.time.LocalDate;

public class ElectronicInvoice extends Invoice {

    ElectronicInvoice(int id, LocalDate installment, Double value, Customer customer) {
        super(id, installment, value, customer);
    }

    //    VIOLATION RULE
//    @Autowired
//    private ClientService clientService;
//
//    public String getFullName() {
//        return clientService.getName();
//    }
    public LocalDate getInstallment() {
        return installment;
    }
}

File: kotlin/spring-arch-unit/spring-arch-unit-mvc-example/src/main/kotlin/com/bmarques/springarchunitmvcexample/domain/invoice/ElectronicInvoice.kt
package com.bmarques.springarchunitmvcexample.domain.invoice

import com.bmarques.springarchunitmvcexample.domain.customer.Customer
import java.time.LocalDate


data class ElectronicInvoice(
        override val id: Int,
        override val installment: LocalDate,
        override val value: Double,
        override val customer: Customer) : Invoice()
